### PR TITLE
Feat: updated dependencies and build script

### PR DESCRIPTION
**Pull request by mrLSD**
## Description
- updated & refactored tests
- refactored mock contracts
- added dummy `mock-eth-connector`
- changed  Engine dependency to PR: split-nep141
- refactored workspace related to new Splitting NEP-141 features
- added build script and extended CI

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ This library is only compatible with Rust version 1.64.0 and above.
 
 ## Tests
 
-Before running tests, build/generate the WASM file as described [here](res/mock_evm/README.md). Then run:
+Before running tests, build/generate the WASM file as described [here](res/mock_engien/README.md). Then run:
```
Comment (aleksuss) [inline README.md:11]:
```suggestion
Before running tests, build/generate the WASM file as described [here](res/mock_engine/README.md). Then run:
```

File: res/mock_engine/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+### How to generate the WASM file
+
+- Install Rust and add the wasm32-unknown-unknown target
+```bash
+rustup target add wasm32-unknown-unknown 
+``` 
+
+- Only for Apple M1/M2 MacOS
+```bash
+rustup default stable-aarch64-apple-darwin
```
Comment (aleksuss) [inline res/mock_engine/README.md:17]: I wonder why it's needed.

File: res/mock_eth_connector/rust-toolchain.toml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[toolchain]
+channel = "1.64.0"
+components = ["rustfmt", "clippy"]
+targets = ["wasm32-unknown-unknown"]
```
Comment (aleksuss) [inline res/mock_eth_connector/rust-toolchain.toml:4]: New line is skipped.

File: workspace/tests/engine_tests.rs [hunk 4]
```diff
@@ -108,3 +109,51 @@ async fn test_register_relayer() {
...
+            "custodian.test.near",
+            FungibleTokenMetadata::default(),
+        )
+        .transact()
+        .await
+        .unwrap();
+}
+/*
+#[tokio::test]
+async fn test_factory_update_address_version() {
```
Comment (aleksuss) [inline workspace/tests/engine_tests.rs:130]: What's wrong with this test?

Comment (mrLSD) [reply workspace/tests/engine_tests.rs:130]: Just forgot remove it.

File: res/mock_engine/README.md (hunk 2 above)
Comment (mrLSD) [reply res/mock_engine/README.md:17]: it's old one.Removed.

File: res/mock_engine/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    // CALLBACK HANDLER METHODS
+    //
+
+    #[result_serializer(borsh)]
+    pub fn factory_update_address_version(&mut self, #[serializer(borsh)] _input: Raw) -> u8 {
+        0
+    }
+
+    #[result_serializer(borsh)]
+    pub fn refund_on_error(&mut self, #[serializer(borsh)] _input: Raw) -> u8 {
```
[the remark this answers is not in the document]
Comment (mrLSD) [reply res/mock_engine/src/lib.rs]: It was old from @mandreyel

File: res/mock_engine/src/fungible_token.rs [hunk 6]
```diff
@@ -1,22 +1,39 @@
...
+            receiver_id,
+            AccountId::from_str("some_account.test").unwrap()
+        );
+        assert_eq!(amount.0, 10);
+        assert_eq!(memo, Some("some message".to_string()));
+    }
+
+    #[allow(unused_variables)]
+    pub fn ft_on_transfer(&mut self, sender_id: AccountId, amount: U128, msg: String) -> String {
+        serde_json::to_string(&0).expect("Failed to serialize message")
```
[the remark this answers is not in the document]
Comment (mrLSD) [reply res/mock_engine/src/fungible_token.rs:24]: I don't know. I just renamed file.

File: workspace/tests/utils.rs [hunk 7]
```diff
@@ -22,7 +22,7 @@ pub async fn init_and_deploy_contract() -> anyhow::Result<EvmContract> {
         owner_id: AccountId::from_str(OWNER_ACCOUNT_ID)?,
         prover_id: AccountId::from_str(PROVER_ACCOUNT_ID)?,
     };
-    let wasm = std::fs::read(WASM_BIN_FILE_PATH)?;
+    let wasm = std::fs::read(WASM_BIN_FILE_PATH).expect("failed read wasm file");
```
Comment (aleksuss) [inline workspace/tests/utils.rs:25]:
```suggestion
    let wasm = std::fs::read(WASM_BIN_FILE_PATH).map_err(|e| anyhow::anyhow!("failed read wasm file: {e}"))?;
```

File: res/mock_engine/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    pub fn set_eth_connector_contract_data(&mut self, #[serializer(borsh)] _input: Raw) {}
+
+    pub fn set_paused_flags(&mut self, #[serializer(borsh)] _input: Raw) {}
+
+    //
+    // CALLBACK HANDLER METHODS
+    //
+
+    #[result_serializer(borsh)]
+    pub fn factory_update_address_version(&mut self, #[serializer(borsh)] _input: Raw) -> u8 {
```
[the remark this answers is not in the document]
Comment (aleksuss) [reply res/mock_engine/src/lib.rs:78]: Could you remove the return type?

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+ENGINE_MOCK_DIR = ${MOCK_DIR}/${ENGINE_DIR}/
+ETH_CONNECTOR_MOCK_DIR = ${MOCK_DIR}/${ETH_CONNECTOR_DIR}/
+MOCK_CARGO_BUILD = RUSTFLAGS='-C link-arg=-s' cargo build --target wasm32-unknown-unknown --release
+
+ENGINE_MOCK_FILE = ${ENGINE_MOCK_DIR}target/wasm32-unknown-unknown/release/mock_engine.wasm
+ETH_CONNECTOR_MOCK_FILE = ${ETH_CONNECTOR_MOCK_DIR}target/wasm32-unknown-unknown/release/mock_eth_connector.wasm
+
+check: check-fmt clippy
+
+clippy: clippy-lib clippy-mock-engine clippy-test
```
Comment (0x3bfc) [inline Makefile:15]: should we include `clippy-mock-eth-connector` as well ?

File: README.md [hunk 10]
```diff
@@ -1,14 +1,29 @@
 # aurora-workspace
 
-Aurora EVM contract for the Near workspaces library
+Aurora Workspace library for the Aurora Engine contract based on NEAR Protocol.
```
Comment (0x3bfc) [inline README.md:3]:
```suggestion
Aurora Workspace is a library for the Aurora Engine and ETH-connector contracts based on NEAR Protocol.
```

File: README.md [hunk 11]
```diff
@@ -1,14 +1,29 @@
...
+- `make check` - run clippy for all code and mock 
+  projects. Check formatting.
+- `make test` - build mock projects and run tests
+- `make test-flow` - run tests only (mock project should be build)
+
+
 ## Tests
 
-Before running tests, build/generate the WASM file as described [here](res/mock_evm/README.md). Then run:
+Before running tests, build/generate the WASM file as described [here](res/mock_engine/README.md).
```
Comment (0x3bfc) [inline README.md:23]: Can we make the makefile generate the wasm file automagically instead of relying on the `README.md` file?

**Review by 0x3bfc — APPROVED** [review]
LGTM

**Review by aleksuss — APPROVED** [review]

**Pull request merged**